### PR TITLE
Changed script and docker-compose files

### PR DESCRIPTION
**Pull request by Turilas83**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ services:
   zabbix-proxy:
     restart: unless-stopped
     image: ${CONTAINER_IMAGE}:${CONTAINER_VERSION}
+    user: 1343
```
Comment (sapajune) [inline docker-compose.yml]: The UID is installation specific so it's not good to hardcode it. Instead there should be default (e.g. 1000) that can be overridden when creating the proxy.

File: create-proxy.sh [hunk 2]
```diff
@@ -38,16 +38,16 @@ echo "Creating container [${NAME}] using image [${CONTAINER_IMAGE}:${CONTAINER_V
...
 if [ -e "${KEY_FILE}" ]; then
-  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${KEY_FILE}\"; chmod 600 \"/var/lib/${KEY_FILE}\"; "
+  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${KEY_FILE}\" ; chmod 600 \"/var/lib/${KEY_FILE}\" ; "
 fi
 if [ -e "${CERT_FILE}" ]; then
-  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${CERT_FILE}\"; chmod 600 \"/var/lib/${CERT_FILE}\"; "
+  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${CERT_FILE}\" ; chmod 600 \"/var/lib/${CERT_FILE}\" ; "
 fi
 
 export COMMAND="${COMMAND}${START_CMD}"
```
Comment (sapajune) [inline create-proxy.sh]: All of the above start command building is unnecessary when permissions are expected to be managed from the host level.

File: create-proxy.sh [hunk 3]
```diff
@@ -25,6 +25,30 @@ KEY_FILE=zabbix/enc/zabbix_proxy_key.pem
...
+
+read -p "Use default group 1000 for GID (Y/n)?" -n 1 -r
+echo
+if [[ ${REPLY} =~ ^[nN]$ ]]; then
+  read -p 'Enter GID: ' group
+else
+  group=1000
+fi
+
+read -p "Use default user 1000 for UID (Y/n)?" -n 1 -r
```
Comment (kompa3) [inline create-proxy.sh]: This is duplicate? Same code twice?

Comment (sapajune) [reply create-proxy.sh]: Looks like it is. Not sure which default UID it should be (100 or 1000). Usually first normal user is 1000 and user's GID matches UID.

Comment (sapajune) [reply create-proxy.sh]: Looking code below updating the permissions the default in the proxy container is 100.

File: create-proxy.sh [hunk 4]
```diff
@@ -37,18 +61,7 @@ echo "Creating container [${NAME}] using image [${CONTAINER_IMAGE}:${CONTAINER_V
...
-if [ -e "${CA_FILE}" ]; then
-  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${CA_FILE}\"; chmod 600 \"/var/lib/${CA_FILE}\"; "
-fi
-if [ -e "${KEY_FILE}" ]; then
-  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${KEY_FILE}\"; chmod 600 \"/var/lib/${KEY_FILE}\"; "
-fi
-if [ -e "${CERT_FILE}" ]; then
-  COMMAND="${COMMAND}chown zabbix:zabbix \"/var/lib/${CERT_FILE}\"; chmod 600 \"/var/lib/${CERT_FILE}\"; "
-fi
+COMMAND="deluser zabbix; addgroup -g ${group} zabbix; adduser -u ${user} -G zabbix -g 'zabbix user' -s /sbin/nologin -D zabbix; find / -user 100 -exec chown zabbix {} \; ; find / -group 1000 -exec chgrp zabbix {} \; ; "
```
Comment (sapajune) [inline create-proxy.sh]: This looks a bit messy solution but I assume it works so I guess it's ok.

**Comment by sapajune** [comment]
> PSK_FILE and other variables are now not used for anything; I wonder whether we should retain the chown commands here just to support PSK and client certificates in case they are used or drop them altogether.

I think PSK files should be no different from client certificate setup and should work in identical way.

File: create-proxy.sh (hunk 3 above)
Comment (Turilas83) [reply create-proxy.sh]: Yeah it's duplicate. I'll remove it. We checked that zabbix users default UID and GID is 100:1000 so changed default values to match that.

**Pull request merged**